src/engine/controls/bpmcontrol.cpp

```diff
@@ -330,12 +330,17 @@ void BpmControl::slotTranslateBeatsHalf(double v) {
     if (!pTrack) {
         return;
     }
-    const double bpm = pTrack->getBpm();
-    const double halfBeat = 3000.0 / bpm;
-    slotTranslateBeatsMove(halfBeat);
+    const mixxx::BeatsPointer pBeats = pTrack->getBeats();
+    if (pBeats) {
+        const auto translatedBeats = pBeats->tryTranslateBeats(0.5);
+        if (translatedBeats) {
+            pTrack->trySetBeats(*translatedBeats);
+        }
+    }
```

Comment:
Stylewise, prefer early returns. It makes the code easier to read because of the reduced nesting.

```suggestion
    const mixxx::BeatsPointer pBeats = pTrack->getBeats();
    if (!pBeats) {
        return;
    }
    const auto translatedBeats = pBeats->tryTranslateBeats(0.5);
    if (!translatedBeats) {
        return;
    }
    pTrack->trySetBeats(*translatedBeats);
```